### PR TITLE
feat(sdk): add timeout option to kfp run submit. Fixes #5806

### LINKED ISSUE #5806: [feature] support timeout option for watching
### Feature Area

/area sdk

### What feature would you like to see?
- When we use `kfp run submit -w`, we have to wait forever when run stay at pending.
- However, I'd like to specify the timeout option with it.

### What is the use case or pain point?
- `kfp run submit -t 60` will wait 60 seconds max, until run be completed
- it would use [kfp client's wait_for_run_completion function](https://github.com/kubeflow/pipelines/blob/6a1b841db923e39016f5c6e66eac8698af27831d/sdk/python/kfp/_client.py#L922)

### Is there a workaround currently?

If you guys agree, I'd like to implement this feature.

---

Love this idea? Give it a 👍. We prioritize fulfilling features with the most 👍.

### PR DESCRIPTION
**Pull request by anencore94**
**Description of your changes:**
Fixes #5806 

add timeout option to `kfp run submit`.
Currently, `kfp run submit` doesn't care about such run finished or not. It is an async request.
Also, 'kfp run submit -w` will stay forever if such run doesn't completed forever.

This pr's feature would be useful to someone who want to get notified just in time when run submit succeeded or failed and who knows approximate completion time of such run.
For those who use `kfp run submit -t <timeout>`, it means something wrong happens if such run won't finish in entered timeout.

- for example
  - `kfp run submit -t 60` will wait 60 seconds max. or finished before 60 seconds if such run's status became one of ['succeeded', 'failed', 'skipped', 'error']

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/cli/run.py [hunk 1]
```diff
@@ -58,10 +58,11 @@ def list(ctx, experiment_id, max_size):
 @click.option('-w', '--watch', is_flag=True, default=False,
               help='Watch the run status until it finishes.')
 @click.option('-v', '--version', help='ID of the pipeline version.')
+@click.option('-t', '--timeout', default=-1, help='Wait for a run to complete until timeout in seconds.')
```
Comment (chensun) [inline sdk/python/kfp/cli/run.py:61]: nit: 0 is probably more common to represent no timeout.

File: sdk/python/kfp/cli/run.py [hunk 2]
```diff
@@ -127,6 +131,15 @@ def _display_run(client, namespace, run_id, watch, output_format):
         _print_runs([run], output_format)
 
 
+def _wait_for_run_completion(client, run_id, timeout, output_format):
+    """
+    wait until run status became one of ['succeeded', 'failed', 'skipped', 'error'],
+    or specified timeout reached
+    """
```
Comment (chensun) [inline sdk/python/kfp/cli/run.py:138]: nit: Given this is only a private method and a very thin wrapper (only 2 lines) over existing methods. I'd rather remove the docstrings.

**Review by chensun — COMMENTED** [review]
/ok-to-test

File: sdk/python/kfp/cli/run.py (hunk 1 above)
Comment (anencore94) [reply sdk/python/kfp/cli/run.py]: Thanks for review! DONE

File: sdk/python/kfp/cli/run.py (hunk 2 above)
Comment (anencore94) [reply sdk/python/kfp/cli/run.py]: I agree :) DONE

File: sdk/python/kfp/cli/run.py [hunk 3]
```diff
@@ -81,7 +82,10 @@ def submit(ctx, experiment_name, run_name, package_file, pipeline_id, pipeline_n
     experiment = client.create_experiment(experiment_name)
     run = client.run_pipeline(experiment.id, run_name, package_file, arg_dict, pipeline_id,
                               version_id=version)
-    _display_run(client, namespace, run.id, watch, output_format)
+    if timeout > 1:
```
Comment (chensun) [inline sdk/python/kfp/cli/run.py:85]: `1` is still a valid timeout. If use ever sets timeout to `1`, the code still needs to behavior correctly, even thought the difference is likely unnoticeable.

Comment (anencore94) [reply sdk/python/kfp/cli/run.py:85]: You are right. Thanks for review!

Comment (anencore94) [reply sdk/python/kfp/cli/run.py]: DONE

**Comment by chensun** [comment]
/lgtm
/approve

Thanks for your contribution!

**Pull request merged**